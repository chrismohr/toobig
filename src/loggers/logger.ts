export interface Result {
  readonly path: string;
  readonly size: number;
  readonly maxSize: number;
}

export interface Logger {
  log(result: Result): void;
  finalize(): Promise<void>;
}
