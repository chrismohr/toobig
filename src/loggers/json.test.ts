import { readFileSync, unlinkSync } from "fs";

import JsonLogger from "./json";
import { Logger } from "./logger";

describe("json logger", () => {
  let filename: string;
  let logger: Logger;

  beforeEach(() => {
    filename = "test-file";
    logger = new JsonLogger(filename);
  });

  afterEach(() => {
    unlinkSync(filename);
  });

  describe(".log", () => {
    describe("with a failure result", () => {
      beforeEach(() => {
        logger.log({ path: "foo", size: 80, maxSize: 100 });
        logger.log({ path: "bar", size: 100, maxSize: 80 });
      });

      describe("when finalized", () => {
        beforeEach(async () => {
          await logger.finalize();
        });

        it("writes the correct results to the file", () => {
          const contents = readFileSync(filename, "utf8");
          const results = JSON.parse(contents);
          expect(results).toMatchSnapshot();
        });
      });
    });

    describe("with only success results", () => {
      beforeEach(() => {
        logger.log({ path: "foo", size: 80, maxSize: 100 });
      });

      describe("when finalized", () => {
        beforeEach(async () => {
          await logger.finalize();
        });

        it("writes the correct results to the file", () => {
          const contents = readFileSync(filename, "utf8");
          const results = JSON.parse(contents);
          expect(results).toMatchSnapshot();
        });
      });
    });
  });
});
