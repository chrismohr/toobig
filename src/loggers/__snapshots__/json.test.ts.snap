// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`json logger .log with a failure result when finalized writes the correct results to the file 1`] = `
Object {
  "failures": Array [
    Object {
      "maxSize": 80,
      "path": "bar",
      "size": 100,
    },
  ],
  "successes": Array [
    Object {
      "maxSize": 100,
      "path": "foo",
      "size": 80,
    },
  ],
}
`;

exports[`json logger .log with only success results when finalized writes the correct results to the file 1`] = `
Object {
  "failures": Array [],
  "successes": Array [
    Object {
      "maxSize": 100,
      "path": "foo",
      "size": 80,
    },
  ],
}
`;
